src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -237,6 +240,71 @@ public Locatable getLocation() {
+     * reduced if they don't). Thus, the algorithm proceeds as following:
+     * <p>
+     * 1. If the bases are the same, the quality of the first element is the sum of both qualities
+     * and the quality of the second is reduced to 0.
+     * 2. If the bases are different, the base with the highest quality is reduced with a factor of
+     * 0.8, and the quality of the lowest is reduced to 0.
+     * <p>
+     * Note: Resulting qualities higher than {@link QualityUtils#MAX_SAM_QUAL_SCORE} are capped.
+     */
+    public void fixOverlaps() {
+        final FragmentCollection<PileupElement> fragments = FragmentCollection.create(this);
+        fragments.getOverlappingPairs().stream()
+                .forEach(
+                        elements -> fixPairOverlappingQualities(elements.get(0), elements.get(1))
+                );
+    }
+
+    /**
+     * Fixes the quality of two elements that come from an overlaping pair in the same way as
+     * samtools does {@see tweak_overlap_quality function in
+     * <a href="https://github.com/samtools/htslib/blob/master/sam.c">samtools</a>}.
+     * The only difference with the samtools API is the cap for high values ({@link QualityUtils#MAX_SAM_QUAL_SCORE}).
+     */
+    @VisibleForTesting
+    static void fixPairOverlappingQualities(final PileupElement firstElement,
+                                            final PileupElement secondElement) {
+        // only if they do not represent deletions
+        if (!secondElement.isDeletion() && !firstElement.isDeletion()) {
+            final byte[] firstQuals = firstElement.getRead().getBaseQualities();
+            final byte[] secondQuals = secondElement.getRead().getBaseQualities();
+            if (firstElement.getBase() == secondElement.getBase()) {
+                // if both have the same base, extra confidence in the firts of them
+                firstQuals[firstElement.getOffset()] =
+                        (byte) (firstQuals[firstElement.getOffset()] + secondQuals[secondElement
+                                .getOffset()]);
+                // cap to maximum byte value
+                if (firstQuals[firstElement.getOffset()] < 0
+                        || firstQuals[firstElement.getOffset()] > QualityUtils.MAX_SAM_QUAL_SCORE) {
+                    firstQuals[firstElement.getOffset()] = QualityUtils.MAX_SAM_QUAL_SCORE;
+                }
+                secondQuals[secondElement.getOffset()] = 0;
+            } else {
+                // if not, we lost confidence in the one with higher quality
+                if (firstElement.getQual() >= secondElement.getQual()) {
+                    firstQuals[firstElement.getOffset()] =
+                            (byte) (0.8 * firstQuals[firstElement.getOffset()]);
+                    secondQuals[secondElement.getOffset()] = 0;
+                } else {
+                    secondQuals[secondElement.getOffset()] =
```

Comment:
Declare the magic value `0.8` as a named `static final` constant at the top of the class, with a name that makes it clear the value comes from samtools, and use the constant in this method instead of the literal value.